File: refactoring-practice-1/src/main/java/dojo/form_template_method/ResidentalSite.java
package dojo.form_template_method;

public class ResidentalSite extends Site {

	public ResidentalSite(int units, double rate) {
		super(units, rate);
	}

	public double getBillableAmount() {
		double base = getUnits() * getRate();
		double tax = base * TAX_RATE;
		return base + tax;
	}
	
}

File: refactoring-practice-1/src/main/typescript/form_template_method/residental-site.ts
import {Site} from "./site.js";

export class ResidentalSite extends Site {

    public constructor(units: number, rate: number) {
        super(units, rate);
    }

    public getBillableAmount() {
        let base = this.getUnits() * this.getRate();
        let tax = base * Site.TAX_RATE;
        return base + tax;
    }

}